backend/shared/shared/azure_adfs/mock_views.py

```diff
@@ -0,0 +1,37 @@
+import logging
+from urllib.parse import urljoin
+
+import factory
+from django.conf import settings
+from django.contrib import auth
+from django.shortcuts import redirect
+from django.views.generic import View
+
+from shared.common.tests.factories import UserFactory
+
+LOGGER = logging.getLogger(__name__)
+
+
+class MockOAuth2LoginView(View):
+    def get(self, request):
+        return redirect("/oauth2/callback?code=mock_adfs_code")
+
+
+class MockOAuth2CallbackView(View):
+    def get(self, request):
+        if not request.user.is_authenticated:
+            user = UserFactory(
+                username=f"handler_{factory.Faker('user_name')}",
+                is_staff=True,
+            )
+            auth.login(
+                request, user, backend="django.contrib.auth.backends.ModelBackend"  # type: ignore
```

Comment:
Why is the `# type: ignore` needed here? Couldn't this be done without the type ignore?

Reply:
This is due to the factory-boy's poor typing support. `user` is `UserFactory` type and `auth.login()` requires basic user object.